From 195bf00a730d5a482f9eff294840c3412efb497e Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Wed, 17 Dec 2025 22:08:25 +0100
Subject: [PATCH 95/96] ia64/mm: Remove redundant VM_*_BIT declarations

Commit 2b6a3f061f11 ("mm: declare VMA flags by bit") declares VMA
flags by bit.

Remove re-definition of bit macros from ia64 code and use VMA_*_BIT
macros instead.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 arch/ia64/mm/fault.c | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/arch/ia64/mm/fault.c b/arch/ia64/mm/fault.c
index 9024a29207b1..eb70dd91195b 100644
--- a/arch/ia64/mm/fault.c
+++ b/arch/ia64/mm/fault.c
@@ -58,10 +58,6 @@ mapped_kernel_page_is_present (unsigned long address)
 	return pte_present(pte);
 }
 
-#	define VM_READ_BIT	0
-#	define VM_WRITE_BIT	1
-#	define VM_EXEC_BIT	2
-
 /* workaround for a warning with -Wmissing-prototypes */
 void __kprobes ia64_do_page_fault (unsigned long address, unsigned long isr, struct pt_regs *regs);
 
@@ -75,8 +71,8 @@ ia64_do_page_fault (unsigned long address, unsigned long isr, struct pt_regs *re
 	vm_fault_t fault;
 	unsigned int flags = FAULT_FLAG_DEFAULT;
 
-	mask = ((((isr >> IA64_ISR_X_BIT) & 1UL) << VM_EXEC_BIT)
-		| (((isr >> IA64_ISR_W_BIT) & 1UL) << VM_WRITE_BIT));
+	mask = ((((isr >> IA64_ISR_X_BIT) & 1UL) << VMA_EXEC_BIT)
+		| (((isr >> IA64_ISR_W_BIT) & 1UL) << VMA_WRITE_BIT));
 
 	/* mmap_lock is performance critical.... */
 	prefetchw(&mm->mmap_lock);
@@ -123,11 +119,6 @@ ia64_do_page_fault (unsigned long address, unsigned long isr, struct pt_regs *re
 
 	/* OK, we've got a good vm_area for this memory area.  Check the access permissions: */
 
-#	if (((1 << VM_READ_BIT) != VM_READ || (1 << VM_WRITE_BIT) != VM_WRITE) \
-	    || (1 << VM_EXEC_BIT) != VM_EXEC)
-#		error File is out of sync with <linux/mm.h>.  Please update.
-#	endif
-
 	if (((isr >> IA64_ISR_R_BIT) & 1UL) && (!(vma->vm_flags & (VM_READ | VM_WRITE))))
 		goto bad_area;
 
-- 
2.25.1

